pkg/osquery/table/app_icons_darwin.go

```diff
@@ -0,0 +1,77 @@
+package table
+
+/*
+#cgo darwin CFLAGS: -DDARWIN -x objective-c
+#cgo darwin LDFLAGS: -framework Cocoa
+#import <Appkit/AppKit.h>
+void Icon(CFDataRef *iconDataRef, char* path) {
+	NSString *appPath = [[NSString stringWithUTF8String:path] stringByStandardizingPath];
+	NSImage *img = [[NSWorkspace sharedWorkspace] iconForFile:appPath];
+
+	//request 512x512 since we are going to resize the icon
+	NSRect targetFrame = NSMakeRect(0, 0, 128, 128);
+	CGImageRef cgref = [img CGImageForProposedRect:&targetFrame context:nil hints:nil];
+	NSBitmapImageRep *brep = [[NSBitmapImageRep alloc] initWithCGImage:cgref];
+	NSData *imageData = [brep TIFFRepresentation];
+	*iconDataRef = (CFDataRef)imageData;
+}
+*/
+import "C"
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"image"
+	"image/png"
+	"unsafe"
+
+	"github.com/kolide/osquery-go/plugin/table"
+	"github.com/nfnt/resize"
+	"github.com/pkg/errors"
+	"golang.org/x/image/tiff"
+)
+
+func AppIcons() *table.Plugin {
+	columns := []table.ColumnDefinition{
+		table.TextColumn("path"),
+		table.TextColumn("icon"),
+	}
+	return table.NewPlugin("kolide_app_icons", columns, generateAppIcons)
+}
+
+func generateAppIcons(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	q, ok := queryContext.Constraints["path"]
+	if !ok || len(q.Constraints) == 0 {
+		return nil, errors.New("The kolide_app_icons table requires that you specify a constraint WHERE path =")
+	}
+	path := q.Constraints[0].Expression
+	img, err := getAppIcon(path)
+	if err != nil {
+		return nil, err
+	}
+
+	var results []map[string]string
+	buf := new(bytes.Buffer)
+	img = resize.Resize(128, 128, img, resize.Bilinear)
```

Comment:
Is this needed? I thought the C code handled the resize?

Also, `resize` looks unmaintained (which might not be terrible, but is a warning to me)

Reply:
CGImageForProposedRect doesn't guarantee the image returned to be of a certain size and only uses the NSRect defined as hints.  

https://developer.apple.com/documentation/appkit/nsimage/1519861-cgimageforproposedrect?language=objc

I grabbed resize from groob's PR didn't notice. Our use case is really simple but I am will to swap it out for an alternative. I does seems to be one of the top used packages for this.

Reply:
Seems fine then. It's what it is.